syntax = "proto3";

message LoginRequest {
  string username = 1;
  string password = 2;
}

message LoginResponse {
  enum Code {
    SUCCESS = 0;
    USER_NOT_FOUND = 1;
    PASSWORD_ERROR = 2;
  }
  Code code = 1;
  int32 uid = 2;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message RegisterResponse {
  enum Code {
    SUCCESS = 0;
    USER_EXISTS = 1;
  }
  Code code = 1;
  int32 uid = 2;
}

message ListUserRequest {
}

message ListUserResponse {
  message User {
    int32 uid = 1;
    string username = 2;
  }
  repeated User users = 1;
}

message Message {
  int32 src_uid = 1;
  int32 dest_uid = 2;
  string msg = 3; // if data exists, msg means file name.
  bytes data = 4;
}
